src/data/finset/basic.lean

```diff
@@ -1447,12 +1453,60 @@ lemma union_sdiff_self (s t : finset α) : (s ∪ t) \ t = s \ t := sup_sdiff_ri
 lemma sdiff_singleton_eq_erase (a : α) (s : finset α) : s \ singleton a = erase s a :=
 by { ext, rw [mem_erase, mem_sdiff, mem_singleton], tauto }
 
-@[simp] lemma sdiff_singleton_not_mem_eq_self (s : finset α) {a : α} (ha : a ∉ s) : s \ {a} = s :=
-by simp only [sdiff_singleton_eq_erase, ha, erase_eq_of_not_mem, not_false_iff]
+lemma erase_eq (s : finset α) (a : α) : s.erase a = s \ {a} := (sdiff_singleton_eq_erase _ _).symm
+
+lemma disjoint_erase_comm : disjoint (s.erase a) t ↔ disjoint s (t.erase a) :=
+by simp_rw [erase_eq, disjoint_sdiff_comm]
+
+lemma disjoint_of_erase_left (ha : a ∉ t) (hst : disjoint (s.erase a) t) : disjoint s t :=
+by { rw [←erase_insert ha, ←disjoint_erase_comm, disjoint_insert_right],
+  exact ⟨not_mem_erase _ _, hst⟩ }
+
+lemma disjoint_of_erase_right (ha : a ∉ s) (hst : disjoint s (t.erase a)) : disjoint s t :=
+by { rw [←erase_insert ha, disjoint_erase_comm, disjoint_insert_left],
+  exact ⟨not_mem_erase _ _, hst⟩ }
+
+@[simp] lemma inter_erase (a : α) (s t : finset α) : s ∩ t.erase a = (s ∩ t).erase a :=
+by simp only [erase_eq, inter_sdiff]
+
+@[simp] lemma erase_inter (a : α) (s t : finset α) : s.erase a ∩ t = (s ∩ t).erase a :=
+by simpa only [inter_comm t] using inter_erase a t s
+
+lemma erase_sdiff_comm (s t : finset α) (a : α) : s.erase a \ t = (s \ t).erase a :=
+by simp_rw [erase_eq, sdiff_right_comm]
+
+lemma insert_union_comm (s t : finset α) (a : α) : insert a s ∪ t = s ∪ insert a t :=
+by rw [insert_union, union_insert]
+
+lemma erase_inter_comm (s t : finset α) (a : α) : s.erase a ∩ t = s ∩ t.erase a :=
+by rw [erase_inter, inter_erase]
+
+lemma erase_union_distrib (s t : finset α) (a : α) : (s ∪ t).erase a = s.erase a ∪ t.erase a :=
+by simp_rw [erase_eq, union_sdiff_distrib]
+
+lemma insert_inter_distrib (s t : finset α) (a : α) : insert a (s ∩ t) = insert a s ∩ insert a t :=
```

Comment:
Can we have the `cons` form of this too?

Reply:
I added it, but is there much point given that we need `decidable_eq α` because of the `∩` anyway?

Reply:
Oh good point, I guess not.

Reply:
I guess it's okay given that `cons` can be useful for keeping the proof around.

Reply:
finset.cons_eq_insert is `simp` so probably it's not effective for that.